hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/TopologySubcommand.java

```diff
@@ -76,6 +89,16 @@ public void execute(ScmClient scmClient) throws IOException {
       if (nodes != null && nodes.size() > 0) {
         // show node state
         System.out.println("State = " + state.toString());
+        if (nodeOperationalState != null) {
+          nodes = nodes.stream().filter(
+              info -> info.getNodeOperationalStates(0).toString()
+                  .equals(nodeOperationalState)).collect(Collectors.toList());
+        }
+        if (nodeState != null) {
```

Comment:
we should add a guardrail here to throw exception if the nodeState isn't HEALTHY/STALE/DEAD.